mlir/include/mlir/Transforms/DialectConversion.h

```diff
@@ -898,6 +898,11 @@ class ConversionPatternRewriter final : public PatternRewriter {
   /// Replace the given operation with the new value ranges. The number of op
   /// results and value ranges must match. The given  operation is erased.
   void replaceOpWithMultiple(Operation *op, ArrayRef<ValueRange> newValues);
+  template <typename RangeT>
+  void replaceOpWithMultiple(Operation *op, RangeT newValues) {
```

Comment:
```suggestion
  void replaceOpWithMultiple(Operation *op, RangeT&& newValues) {
```
this would trigger coying of `SmallVector`s and similar otherwise